VimR/Workspace/Workspace.swift

```diff
@@ -158,20 +169,34 @@ extension Workspace {
   }
 
   override func draggingExited(_ sender: NSDraggingInfo?) {
-    self.endDrag()
+    self.endDrag(sender: sender)
   }
 
   override func draggingEnded(_ sender: NSDraggingInfo?) {
-    self.endDrag()
+    self.endDrag(sender: sender)
   }
 
-  fileprivate func endDrag() {
+  fileprivate func endDrag(sender: NSDraggingInfo?) {
     self.isDragOngoing = false
+    if isFile(sender: sender) {
+        return
+    }
     self.draggedOnBarLocation = nil
     self.proxyBar.removeFromSuperview()
   }
 
+  fileprivate func isFile(sender: NSDraggingInfo?) -> Bool! {
+    return (sender?.draggingPasteboard().types?.contains(String(kUTTypeFileURL)))!
+  }
+  
   override func performDragOperation(_ sender: NSDraggingInfo) -> Bool {
+    if isFile(sender: sender) {
+      let paths = sender
+        .draggingPasteboard()
+        .propertyList(forType: "NSFilenamesPboardType") as? [String]
```

Comment:
Maybe if we use `NSPasteboard.readObjects(forClasses:options:)`, we don't have to use the a string constant for type? But I'm not sure...